### ISSUE TITLE
Using backend for both React Native and Custom Admin page with React JS

### ISSUE THREAD

**Issue by fistofzen** [aws-amplify/amplify-js#13397]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React, React Native

### Amplify APIs

Authentication, REST API, GraphQL API, Storage

### Amplify Version

v6

### Amplify Categories

auth, storage, function, api

### Backend

None

### Environment information

  System:
    OS: macOS 14.3.1
    CPU: (8) arm64 Apple M3
    Memory: 76.47 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.1 - /usr/local/bin/node
    Yarn: 1.22.21 - /usr/local/bin/yarn
    npm: 10.2.4 - /usr/local/bin/npm
  Browsers:
    Chrome: 124.0.6367.208
    Safari: 17.3.1
  npmGlobalPackages:
    @aws-amplify/cli: 12.12.0

### Describe the bug

I created a backend with using amplify on a react native app .
Then I created another React JS app which is an admin for this react native app. they are using the same backend. 

I deployed the app to the aws but I want only admins to login to the custom react js admin page. I am using withAuthenticator but I didnt find an option about cognito groups. 

End users will use React native and admins will use admin app. 

how can I achieve that ? 

### Expected behavior

prevent normal users to login to the custom admin website.

### Reproduction steps

reploy react js app to the aws
login with admin user

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by israx** [comment]
hello @fistofzen. A potential solution is to create different `app client ids` for your userpool and assign custom scopes to each one. You need to authenticate with either hosted UI or a 3red IdP in other to get these custom scopes in your access token.

Bellow are the steps on how to add custom scopes to an app client id

1. Go to the `App clients and analytics` section and create one app client id for `web` and other for `react native`. You can find this section under the `App integration` tab.
2. Go to the `Resource servers` section and create a [resource](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-define-resource-servers.html#cognito-user-pools-define-resource-servers-about-resource-servers) server for your userpool and assign some custom scopes.
3. Go to the created `app client id` and go to `Hosted UI` section, and set up OAuth settings.
4. assign the app client ids to your apps

Remember that you need to authenticate with hosted UI or configure a different IdP provider in order to get those scopes in your access tokens.

**Comment by fistofzen** [comment]
Hello israx, Thank you for the answer.
How can I assign the app client ids to your apps ? and also,  how can I set up OAuth settings? 

Regards.

**Comment by fistofzen** [comment]
actually I have A user which is assigned to a group "Admins". And I want only that user to successfully sign in to the admin app. React Native and ReactJS is using same identity pool.

**Comment by fistofzen** [comment]
> hello @fistofzen. A potential solution is to create different `app client ids` for your userpool and assign custom scopes to each one. You need to authenticate with either hosted UI or a 3red IdP in order to get these custom scopes in your access token.
> 
> Bellow are the steps on how to add custom scopes to an app client id
> 
> 1. Go to the `App clients and analytics` section and create one app client id for `web` and other for `react native`. You can find this section under the `App integration` tab of your userpool.
> 2. Go to the `Resource servers` section and create a [resource](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-define-resource-servers.html#cognito-user-pools-define-resource-servers-about-resource-servers) server for your userpool and assign some custom scopes. You can find this section under the `App integration` tab of your userpool.
> 3. Go to the created `app client id` and go to `Hosted UI` section, and set up OAuth settings.
> 4. assign the app client ids to your apps
> 
> Remember that you need to authenticate with hosted UI or configure a different IdP provider in order to get those scopes in your access tokens.

When I go with this option I am getting error 

![image](https://github.com/aws-amplify/amplify-js/assets/6533119/7eae6bbb-0b89-42d4-a9f8-a8373858eaf1)

**Comment by israx** [comment]
hello @fistofzen can you provide the following info ?

1. How did you configure hosted UI. Can you show a screenshoot of the hosted ui settings you setup ?
2. How are you login ? are you using the `signInWithRedirect` API ?
3. How are you configuring Amplify for RN and Web ? Please attach some code snippets for both configurations without showing sensitive values
4. What API is throwing that error you attached above ? Seems you are using graphql and getting `UnAuthorized` exceptions. Please make sure to associate your userpool with your identity pool

**Comment by israx** [comment]
Regarding @biller-aivy comment, you can setup a `pre-token` lambda but you need to enable `Advance Security` features which will increase your price per user exponentially

**Comment by fistofzen** [comment]
Hello İsrax,
1 - 
<img width="780" alt="image" src="https://github.com/aws-amplify/amplify-js/assets/6533119/4173d5b5-d50c-45aa-ad59-bea2ff848eb2">

2 - I am not using signInWithRedirect,
3 - 
<img width="780" alt="image" src="https://github.com/aws-amplify/amplify-js/assets/6533119/40e5e99e-992e-4494-a810-b336a36c4433">
4 - yes graphql. 

Here I only want to signin users who are in admin group. no others.

**Comment by fistofzen** [comment]
<img width="780" alt="image" src="https://github.com/aws-amplify/amplify-js/assets/6533119/0b9da58b-921e-4c0f-b394-942703a8323d">

**Comment by israx** [comment]
hello @fistofzen . In order to get the custom scope in your access token. You need to configure OAuth with Amplify — We have some [documentation](https://docs.amplify.aws/gen1/react/build-a-backend/auth/add-social-provider/#redirect-urls) about that. Then you need to make sure to call the `signInWithRedirect` API.

**Comment by cwomack** [comment]
@fistofzen, wanted to ping you to see if you had a chance to review @israx's comment above.  Let us know if you're still blocked by this! Thanks.

**Comment by fistofzen** [comment]
Yes I am still blocked .

**Comment by ashika112** [comment]
@fistofzen Apolozies on getting back on this earlier? Where are you blocked right now after @israx last suggestion?

**Comment by ashika112** [comment]
Closing this since we have not heard back from you. If this is still an issue or have any further ones, please open a new issue.
